src/platform/Linux/MdnsImpl.cpp

```diff
@@ -605,8 +605,13 @@ void MdnsAvahi::HandleBrowse(AvahiServiceBrowser * browser, AvahiIfIndex interfa
 
             Platform::CopyString(service.mName, name);
             CopyTypeWithoutProtocol(service.mType, type);
-            service.mProtocol               = GetProtocolInType(type);
-            service.mAddressType            = ToAddressType(protocol);
+            service.mProtocol    = GetProtocolInType(type);
+            service.mAddressType = ToAddressType(protocol);
+            service.mInterface   = INET_NULL_INTERFACEID;
+            if (interface != AVAHI_IF_UNSPEC)
+            {
+                service.mInterface = static_cast<chip::Inet::InterfaceId>(interface);
```

Comment:
could you add some documentation on why this cast is ok? What guarantees that the avahi types and chip inet types are the same and why is a cast needed (if they are the same types I would expect just equality to work)?

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/9526

Reply:
@woody-apple why could we not have added the comment about why cast is needed and why it is safe in this PR?